### PR TITLE
Update policy and dynamic-config APIs

### PR DESCRIPTION
**Pull request by hasit**
### Description of change

- Update etcd calls to be blocking using `etcdClient` directly
- Move away from adding/updating multiple policies and dynamic-configs in a single call
- Combine a POST and PATCH into 1 call, by using `google.protobuf.FieldMask`, still keeping both methods and backwards compatibility with `aperturectl`
- Add FieldMask related helper functions in utils
- Replace "no matches for kind" string check with `k8s.io/apimachinery/pkg/api/meta` errors check

##### Checklist

- [x] Tested in playground or other setup
- [ ] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

### Summary by OpenAI

**New Feature:**
- Combine POST and PATCH into a single call using `google.protobuf.FieldMask`
- Add FieldMask helper functions in the `utils` package

**Refactor:**
- Update etcd calls to be blocking
- Simplify policy and dynamic-config updates by separating them
- Improve error handling with `k8s.io/apimachinery/pkg/api/meta` errors check

> 🎉 A merge of POST and PATCH we see, 🔄
> With FieldMask helpers, oh so nifty! 🛠️
> Etcd calls now block, as they should, ⏱️
> And error checks improved, all understood. 🚀

### CODE REVIEW

**Review by harjotgill — APPROVED** [review]

File: pkg/cmd/controller.go [hunk 1]
```diff
@@ -351,19 +351,14 @@ agentsLoop:
...
+func (h *Handler) UpsertPolicy(ctx context.Context, req *policylangv1.UpsertPolicyRequest) (*emptypb.Empty, error) {
+	return h.policyService.UpsertPolicy(ctx, req)
 }
 
 // PostDynamicConfigs updates dynamic-configs to the system.
-func (h *Handler) PostDynamicConfigs(ctx context.Context, req *policylangv1.PostDynamicConfigsRequest) (*policylangv1.PostResponse, error) {
-	return h.policyService.PostDynamicConfigs(ctx, req)
+func (h *Handler) PostDynamicConfig(ctx context.Context, req *policylangv1.PostDynamicConfigRequest) (*emptypb.Empty, error) {
+	return h.policyService.PostDynamicConfig(ctx, req)
 }
```
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/cmd/controller.go]: Addressed in [6cf6972](https://github.com/fluxninja/aperture/pull/1767/commits/6cf697236726106589a3b38c5ac2c7f480266656)

File: cmd/aperturectl/cmd/apply/policy.go [hunk 2]
```diff
@@ -199,7 +192,10 @@ func updatePolicyUsingAPI(name string, policy *languagev1.Policy) (bool, error)
...
 			}
 
-			_, err = client.PatchPolicies(context.Background(), &request)
+			request.UpdateMask = &field_mask.FieldMask{
+				Paths: []string{"all"},
+			}
+			_, err = client.UpsertPolicy(context.Background(), &request)
 			if err != nil {
 				return false, err
 			}
```
[the remark this answers is not in the document]
Comment (hasit) [reply cmd/aperturectl/cmd/apply/policy.go:201]: Addressed in [6409601](https://github.com/fluxninja/aperture/pull/1767/commits/6409601ba676b08749bc816b3f03bb8f58c3e555)

File: pkg/policies/controlplane/policy-service.go [hunk 3]
```diff
@@ -72,35 +74,65 @@ func (s *PolicyService) GetPolicy(ctx context.Context, request *policylangv1.Get
...
+		return nil, err
+	}
+
+	_, err = s.etcdClient.Client.KV.Put(ctx, path.Join(paths.PoliciesAPIConfigPath, req.PolicyName), string(jsonPolicy))
+	if err != nil {
+		return nil, fmt.Errorf("failed to write policy '%s' to etcd: '%s'", req.PolicyName, err)
+	}
+
+	return new(emptypb.Empty), nil
+}
```
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/policies/controlplane/policy-service.go]: Addressed in [12aa2f1](https://github.com/fluxninja/aperture/pull/1767/commits/12aa2f16caa456a07d82a1272ec22f6efccdbc88)

File: pkg/policies/controlplane/policy-service.go [hunk 4]
```diff
@@ -72,35 +74,65 @@ func (s *PolicyService) GetPolicy(ctx context.Context, request *policylangv1.Get
...
+func (s *PolicyService) UpsertPolicy(ctx context.Context, req *policylangv1.UpsertPolicyRequest) (*emptypb.Empty, error) {
+	if req.UpdateMask == nil && len(req.UpdateMask.GetPaths()) == 0 {
+		return s.upsertPolicy(ctx, req, false)
+	}
 
-// PatchPolicies patches policies to the system.
-func (s *PolicyService) PatchPolicies(ctx context.Context, policies *policylangv1.PostPoliciesRequest) (*policylangv1.PostResponse, error) {
-	return s.updatePolicies(ctx, policies, true)
+	return s.upsertPolicy(ctx, req, true)
 }
```
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/policies/controlplane/policy-service.go]: Addressed in [12aa2f1](https://github.com/fluxninja/aperture/pull/1767/commits/12aa2f16caa456a07d82a1272ec22f6efccdbc88)

**Pull request merged**